swap_meet/vendor.py

```diff
@@ -1,2 +1,88 @@
+from unicodedata import category
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory = None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory 
+            
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+
+    def get_by_category(self, category):
+        if not category:
+            return False
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list 
+
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory:
+            return False
+        if their_item not in friend.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+        friend.inventory.append(my_item) 
```

Comment:
`remove` and `append` are being called directly on `self.inventory` and `friend.inventory`, but we could also use the `add` and `remove` functions we wrote here:
```suggestion
        self.remove(my_item)
        friend.add(my_item) 
```